jyulb/internal/common/tests/test_proxy_lattice.py

```diff
@@ -0,0 +1,133 @@
+
+        proxy.data = self.hl_data
+
+        self.assertIsInstance(proxy, ABCLattice, "Error: not a ABCLattice!")
+
+        self.assertEqual(proxy.name, self.name)
+        self.assertEqual(proxy.type, self.type)
+        self.assertEqual(proxy._geom, self.geom)
+        self.assertEqual(proxy._fdata, self.fdata)
+        self.assertEqual(proxy.data, self.hl_data)
+        np_test.assert_array_equal(proxy.size, self.size)
+        np_test.assert_array_equal(proxy.origin, self.orig)
+        np_test.assert_array_equal(proxy.base_vect, self.bvec)
+
+        try:
+            proxy.get_node((-1, 0, 0))
+        except IndexError:
+            pass
+        else:
+            raise AssertionError('Negative indices incorrectly accepted.')
+
+        test_node = LatticeNode((0, -1, 0))
+        try:
+            proxy.update_node(test_node)
+        except IndexError:
+            pass
+        else:
+            raise AssertionError('Negative indices incorrectly accepted.')
+
+    def test_set_get_iter_lattice_nodes(self):
+        """Creation of lattices using the factory functions."""
+        proxy = ProxyLattice(self.name, self.type,
+                             self.bvec, self.geom, self.fdata)
+
+        iden = 1.0
+        ivel = np.array((0.0, 0.0, 1.0), dtype=np.float64)
+        ifrc = np.array((0.0, 1.0, 0.0), dtype=np.float64)
+
+        for node in proxy.iter_nodes():
+            node.data[CUBA.DENSITY] = iden
+            node.data[CUBA.VELOCITY] = ivel
+            node.data[CUBA.FORCE] = ifrc
+            proxy.update_node(node)
+
+        check_iden = 0.0
+        check_ivelx = 0.0
+        check_ively = 0.0
+        check_ivelz = 0.0
+        check_ifrcx = 0.0
+        check_ifrcy = 0.0
+        check_ifrcz = 0.0
+        check_sum = (self.nx-2)*self.ny*self.nz
+
+        for node in proxy.iter_nodes():
+            if node.data[CUBA.MATERIAL_ID] == ProxyLattice.FLUID_ENUM:
+                check_iden = check_iden + node.data[CUBA.DENSITY]
+                check_ivelx = check_ivelx + node.data[CUBA.VELOCITY][0]
+                check_ively = check_ively + node.data[CUBA.VELOCITY][1]
+                check_ivelz = check_ivelz + node.data[CUBA.VELOCITY][2]
+                check_ifrcx = check_ifrcx + node.data[CUBA.FORCE][0]
+                check_ifrcy = check_ifrcy + node.data[CUBA.FORCE][1]
+                check_ifrcz = check_ifrcz + node.data[CUBA.FORCE][2]
+
+        self.assertEqual(check_sum, check_iden)
+        self.assertEqual(0.0, check_ivelx)
+        self.assertEqual(0.0, check_ively)
+        self.assertEqual(check_sum, check_ivelz)
+        self.assertEqual(0.0, check_ifrcx)
+        self.assertEqual(check_sum, check_ifrcy)
+        self.assertEqual(0.0, check_ifrcz)
+
+        for node in proxy.iter_nodes(np.ndindex(1, self.ny, self.nz)):
+            try:
+                node.data[CUBA.DENSITY]
+                node.data[CUBA.VELOCITY]
+                node.data[CUBA.FORCE]
+            except:
```

Comment:
This will silently ignore any errors raised. If I am correct it asserts that there is an exception raised but the check is not clear on what it tries to achieve.